### ISSUE TITLE
[naming-convention] Naming convention rule should allow more options  

### ISSUE THREAD

**Issue by eddeee888** [dotansimha/graphql-eslint#195]
Currently, the only thing this rule allows is checking format which is a very good start!

I think it would be even better if we could allow other types of checks such as:

- suffix e.g. Inputs must have `Input` suffix: `CreateMessageInput`
- allows special characters. e.g Fragments must be Pascal case but can have special characters in between: `CreateUser_User`

One way I think we could do it is allowing an object instead of strings for the rule e.g.

```
"@graphql-eslint/naming-convention": [
          "error",
          {
            "InputObjectTypeDefinition": {
               "format": "PascalCase",
               "suffix": "Input"
            },
            "FragmentDefinition": { 
               "format": "PascalCase",
               "allowSpecialCharacters": "_-"
          }
        ]
```

Alternatively, maybe we could allow passing custom regex?

If this makes sense, I can try creating a PR for it :)

**Comment by dotansimha** [comment]
Thank you @eddeee888! 

For inputs name, you can use this rule: https://github.com/dotansimha/graphql-eslint/blob/master/docs/rules/input-name.md 

Regarding the rest of the things you mentioned, I think we can add that to the rule itself as well, to allow more flexibility.

PRs are always welcome!

**Comment by eddeee888** [comment]
> you can use this rule: https://github.com/dotansimha/graphql-eslint/blob/master/docs/rules/input-name.md

Hi @dotansimha, I tried using the [input name rule]( https://github.com/dotansimha/graphql-eslint/blob/master/docs/rules/input-name.md) but it currently has some tight restrictions ( maybe it needs its own issue ):
- It's only being applied to mutations, but users may want to use it for queries as well
- It's forcing the case of the input type e.g.
```graphql
type Mutation {
  postMessage(input: PostMessageInput!): Message! 
}
```
In this example, it fails because it wants the input type to be `postMessageInput` instead of `PostMessageInput`. However, it might conflict with the `naming-convention` rule if we set `InputObjectTypeDefinition` to `PascalCase`


> Regarding the rest of the things you mentioned, I think we can add that to the rule itself to allow more flexibility.

By "add that to the rule itself", do you mean I should update this [file](https://github.com/dotansimha/graphql-eslint/blob/master/packages/plugin/src/rules/naming-convention.ts)?
Maybe an example of how the config would look if it's different from what I posted? :)

**Comment by eddeee888** [comment]
@dotansimha An early draft of the PR for this issue is done here: https://github.com/dotansimha/graphql-eslint/pull/201. Please let me know if it makes sense 🙏

**Comment by dotansimha** [comment]
Available in `@graphql-eslint/eslint-plugin@0.7.0`. Thanks @eddeee888
